Reference_Manual.md

```diff
@@ -3287,6 +3294,245 @@ This typed_data allows you to retrieve the data type of the underlying binary to
+        auto page = page_f.get();
+    }
+}
+catch (const hazelcast_sql_exception&)
+{
+    // Do error handling
+}
+```
+
+### 7.11.4 Sql Statement With Options
+`sql_statement` is a class which holds query string, its parameters and options.
+It can be passed to `sql_service::execute` method and its parameters and options will be taken into consideration.
+
+``` C++
+using namespace hazelcast::client::sql;
+
+auto hz = hazelcast::new_client().get();
+
+// Constructs sql_statement with default options which are :
+// cursor_buffer_size: 4096
+// timeout: std::chrono::millisecond{ -1 } which means no timeout
+// expected_result_type: any
+sql_statement statement
+{
+    hz ,
+    R"(
+        SELECT * FROM some_table WHERE this > ?
+    )"
+};
+
+// Methods which specifies options are able to be called in chain, returns self references
+statement.add_parameter(15) // '?' part of the query will be filled with this parameter.
+         .cursor_buffer_size(96) // Set '96' to 'cursor_buffer_size' option
+         .timeout(std::chrono::milliseconds {500}) // Set '500' milliseconds timeout, if the time is exceeded query will be cancelled.
+         .expected_result_type(sql_expected_result_type::rows); // Expect a table which contains rows
+
+// To fetch use non-parameterized overloads
+std::cout << "cursor_buffer_size : "   << statement.cursor_buffer_size()        << std::endl
+          << "timeout : "              << statement.timeout().count()           << std::endl
+          << "expected_result_type : " << int(statement.expected_result_type()) << std::endl;
+
+hz.get_sql().execute(statement); // OK, execute this statement
+```
+
+### 7.11.5 Read Row Metadata
+Row metadatas can be acquired with `sql_result::row_metadata()` and `sql_page::sql_row::row_metadata()` method.
+
+``` C++
+using namespace hazelcast::client::sql;
+
+auto hz = hazelcast::new_client().get();
+
+auto result = hz.get_sql().execute("SELECT name, surname, age FROM employees").get();
+
+auto metadata = result->row_metadata();
+
+// Print column count
+std::cout << "There are " << metadata.column_count() << " columns" << std::endl;
+
+// Print columns
+for (const sql_column_metadata& col : metadata.columns())
+{
+    std::cout << "name : "      << col.name
+              << " type : "     << int(col.type)
+              << " nullable : " << col.nullable
+              << std::endl;
+}
+
+// Get column by index
+auto age_col_metadata = metadata.column(2);
+
+std::cout << "name : "      << age_col_metadata.name
+          << " type : "     << int(age_col_metadata.type)
+          << " nullable : " << age_col_metadata.nullable
+          << std::endl;
+```
+**Note :** There is no metadata for non-`SELECT` queries.
+
+``` C++
+auto result = sql.execute("DELETE FROM employees WHERE age < 18").get();
+
+result->row_metadata(); // Throws an `exception::illegal_state`
+                        // This is not a SELECT query
+```
+
+### 7.11.6 Read Cell Value
```

Comment:
What is a Cell? Sql Table Cell?

Reply:
I believe we usually use the term column value instead of table cell. did you see any place in server docs etc. mention about table cell? can we double check on this?

Reply:
fixed